### DISCUSSION TITLE
Determine bar height in a tooltip's custom cursor

### DISCUSSION THREAD

**Discussion by jannikbertram** [recharts/recharts#3777] [Q&A]
I'm trying to build a bar chart with custom styling on hover. For that I use a Tooltip with a custom component as cursor:
```
export const HighlightBarCursor = ({
  height,
  points,
  payload,
  barSize,
  maxDataValue
}: RechartsCursorProps) => {
  const { x, y } = points[1];
  const dataValue = payload[0].value;

  const barHeight = maxDataValue
    ? (dataValue / maxDataValue) * height
    : height;
  
  // Without that extra margin, the rectangle's border overlaps with the bars
  const extraMargin = 0;

  return (
    <Rectangle
      className="drop-shadow-xl"
      fill="white"
      x={x - Math.ceil(barSize / 2) - extraMargin}
      y={y}
      width={barSize + extraMargin}
      height={barHeight + extraMargin}
      stroke="black"
      strokeOpacity={0.5}
    />
  );
};
```

The tooltip using the cursor looks like this (cursor props like `height`, and `points` are automatically passed to the custom cursor):
```
          <Tooltip
            cursor={
              <HighlightBarCursor
                barSize={BAR_SIZE}
                maxDataValue={maxDataValue}
              />
            }
          />
```
For a bar chart with just one bar, this works fine. However, with a more evenly distributed chart, the height of the each bar is not computed correctly

![Screenshot 2023-09-22 at 13 58 17](https://github.com/recharts/recharts/assets/17082165/c75058e5-6822-4d91-851e-c439c708e1a9)

![Screenshot 2023-09-22 at 13 58 36](https://github.com/recharts/recharts/assets/17082165/61d25955-9090-49ad-8c65-be0579af23a1)

Is there a way to compute the correct `y` coordinate for a bar given the chart's `height`, the bar's `dataValue`, and the 
`maxDataValue`? 

Any help would be appreciated! Cheers

---

注意：不是用 https://recharts.github.io/recharts-issue-helper 创建的 issue 会被立即关闭。

**Comment by ckifer** [comment]
Hi @jannikbertram - we just merged some changes to add an activeBar on hover. If you can wait a few weeks or so that will be out and should solve your case without any extra work

**Reply by jannikbertram** [reply]
Cool, I will watch out for a new version with that feature. Looking forward to it!
